test/libsolidity/SolidityParser.cpp

```diff
@@ -1479,6 +1479,20 @@ BOOST_AUTO_TEST_CASE(function_type_state_variable)
 	BOOST_CHECK(successParse(text));
 }
 
+BOOST_AUTO_TEST_CASE(scientific_notation)
+{
+	char const* text = R"(
+		contract test {
+			uint256 a = 2e10;
```

Comment:
Sorry, I cannot find it. I mean a number with negative mantissa, not negative exponent.